### PR TITLE
test: use `tmp_path` fixture instead of `tmpdir` 

### PR DESCRIPTION
**Pull request by miltolstoy**

### CODE REVIEW

File: tests/deployment_utils.py [hunk 1]
```diff
@@ -154,26 +155,27 @@ def clean_execution_environment(request):
...
         except OSError:
             pass
-        source_dir.ensure(dir=True)
+        source_dir.mkdir()
 
     else:
-        source_dir = tmpdir.mkdir("project_sources")
-    local_file = source_dir.join("readme.txt")
-    local_file.write("This is a an empty file")
+        source_dir = tmp_path.joinpath("project_sources")
```
Comment (i-keliukh) [inline tests/deployment_utils.py:175]: Wouldn't it be better to write `tmp_path / "project_sources"` instead?

File: tests/deployment_utils.py [hunk 2]
```diff
@@ -165,7 +166,7 @@ def local_sources(tmp_path: py.path.local):
     if utils.reuse_docker_containers():
         source_dir = py.path.local(".work")
         try:
-            source_dir.remove(rec=1, ignore_errors=True)
+            shutil.rmtree(str(source_dir), ignore_errors=True)
```
Comment (i-keliukh) [inline tests/deployment_utils.py:169]: Cast to str is not needed

File: tests/test_p4_exception_handling.py [hunk 3]
```diff
@@ -27,7 +27,7 @@ def test_p4_forbidden_local_revert(perforce_environment: P4TestEnvironment, stdo
     perforce_environment.shelve_config(config)
     result = __main__.run(perforce_environment.settings)
     # Clean up the directory at once to make sure it doesn't remain non-writable even if some assert fails
-    perforce_environment.temp_dir.chmod(0o0777, rec=1)
+    perforce_environment.temp_dir.chmod(0o0777)
```
Comment (i-keliukh) [inline tests/test_p4_exception_handling.py]: It was recursive in previous version. Not sure whether it is critical or not.

Comment (miltolstoy) [reply tests/test_p4_exception_handling.py]: Was fixed in [one](https://github.com/Samsung/Universum/pull/757/commits/0ed013fff73236a720a7028f6958f4cf732f4fb4) of the following commits

File: tests/deployment_utils.py [hunk 4]
```diff
@@ -2,10 +2,11 @@
 
 import getpass
 import os
+import shutil
+import pathlib
```
Comment (k-dovgan) [inline tests/deployment_utils.py:6]: We have several places in repo where `from pathlib import Path` is used. Maybe we want to standardize Path usage in this very commit?

File: tests/test_check_if_env_set.py [hunk 5]
```diff
@@ -1,7 +1,7 @@
 # pylint: disable = redefined-outer-name
 
 import os
-from typing import Optional
+from typing import Optional, List
```
Comment (k-dovgan) [inline tests/test_check_if_env_set.py:4]: Why did it work before this change?

**Review by i-keliukh — APPROVED** [review]

**Pull request merged**